pkg/nfd-master/nfd-master.go

```diff
@@ -519,24 +519,62 @@ func (m *nfdMaster) filterFeatureLabels(labels Labels) (Labels, ExtendedResource
 	return outLabels, extendedResources
 }
 
-func (m *nfdMaster) filterFeatureLabel(name string) error {
+func (m *nfdMaster) filterFeatureLabel(name, value string, features *nfdv1alpha1.Features) (string, error) {
 	// Check label namespace, filter out if ns is not whitelisted
 	ns, base := splitNs(name)
 	if ns != nfdv1alpha1.FeatureLabelNs && ns != nfdv1alpha1.ProfileLabelNs &&
 		!strings.HasSuffix(ns, nfdv1alpha1.FeatureLabelSubNsSuffix) && !strings.HasSuffix(ns, nfdv1alpha1.ProfileLabelSubNsSuffix) {
 		// If the namespace is denied, and not present in the extraLabelNs, label will be ignored
 		if isNamespaceDenied(ns, m.deniedNs.wildcard, m.deniedNs.normal) {
 			if _, ok := m.config.ExtraLabelNs[ns]; !ok {
-				return fmt.Errorf("namespace %q is not allowed", ns)
+				return "", fmt.Errorf("namespace %q is not allowed", ns)
 			}
 		}
 	}
 
 	// Skip if label doesn't match labelWhiteList
 	if !m.config.LabelWhiteList.Regexp.MatchString(base) {
-		return fmt.Errorf("%s (%s) does not match the whitelist (%s)", base, name, m.config.LabelWhiteList.Regexp.String())
+		return "", fmt.Errorf("%s (%s) does not match the whitelist (%s)", base, name, m.config.LabelWhiteList.Regexp.String())
 	}
-	return nil
+
+	// Dynamic Value
+	if strings.HasPrefix(value, "@") {
+		if value, err := getDynamicValue(value, features, false); err != nil {
+			return "", err
+		} else {
+			return value, nil
+		}
+	}
+
+	return value, nil
+}
+
+func getDynamicValue(value string, features *nfdv1alpha1.Features, parseQuantity bool) (string, error) {
```

Comment:
I'd suggest to leave the validation out of this func, just to make it cleaner

```suggestion
func getDynamicValue(value string, features *nfdv1alpha1.Features) (string, error) {
```

or then add a generic validation but I don't know about this either


```suggestion
func getDynamicValue(value string, features *nfdv1alpha1.Features, validate func(string) error) (string, error) {
```